### PR TITLE
Add {column}_colors

### LINKED ISSUE #216: do or do not (there is no try)  _colors
[`_colors` ](https://github.com/chanzuckerberg/single-cell-curation/blob/main/schema/1.1.0/corpora_schema.md#presentation-hints)was not included in schema 2.0.0 because it was perceived as an [under documented feature](https://github.com/chanzuckerberg/cellxgene-documentation/blob/main/desktop/data-reqs.md#data-format-options) of cellxgene desktop and was not been present in most submissions in the portal at the time. There was no validation performed by `cellxgene-schema CLI 1.1.0` or in the subsequent cellxgene conversion so support was brittle.

Recently, there was a CXG conversion failure during data ingestion reported in [**#single-cell-data-wrangling**](https://czi-sci.slack.com/archives/C024HCSH9PT/p1651072467672089) due to the following causes:

1. at least one color table existed for a non-existent obs column 
1. at least one color table existed for a non-categorical obs column

`_colors` should either be blocked+fail or documented+validated in the future.

### PR DESCRIPTION
**Pull request by brianraymor**
Closes #216

### CODE REVIEW

File: schema/4.0.0/schema.md [hunk 1]
```diff
@@ -1112,6 +1112,71 @@ See also `default_embedding` in `uns`.
...
 <br>
 
+
+### {category}_colors
+
+<table><tbody>
+  <tr>
+    <th>Key</th>
+      <td>
+        {category}_colors where {category} MUST be the name of a <code>category</code> data type column in <code>obs</code> that<br> is annotated by the data submitter or curator. The following columns that are annotated by CELLxGENE<br> Discover MUST NOT be specified as {category}:<br><br>
```
Comment (jahilton) [inline schema/4.0.0/schema.md:1122]: The multi-use of `category` here made me have to think really hard.
I think using `column` when referring to the obs field name would alleviate this, and be consistent with the rest of the schema (e.g. "Curators MUST annotate the following _columns_...")

File: schema/4.0.0/schema.md [hunk 2]
```diff
@@ -1112,6 +1112,71 @@ See also `default_embedding` in `uns`.
...
+      <ul>
+        <li>assay</li>
+        <li>cell_type</li>
+        <li>development_stage</li>
+        <li>disease</li>
+        <li>organism</li>
+        <li>self_reported_ethnicity</li>
+        <li>sex</li>
+        <li>tissue</li>
```
Comment (jahilton) [inline schema/4.0.0/schema.md:1131]: I feel like I'm left hanging with a "but what if I want to define the palette for one of those fields?"
After the list add "Instead, annotate {category}_ontology_term_id_colors for those columns"?

File: schema/4.0.0/schema.md (hunk 1 above)
Comment (brianraymor) [reply schema/4.0.0/schema.md]: Addressed in b1a41adbfffc1a5ee1280deeb13fec4460abad8a.

File: schema/4.0.0/schema.md (hunk 2 above)
Comment (brianraymor) [reply schema/4.0.0/schema.md:1131]: Addressed in b1a41adbfffc1a5ee1280deeb13fec4460abad8a.

Comment (brianraymor) [reply schema/4.0.0/schema.md:1131]: @jahilton - I could also create a table if that would be easier to comprehend:

| **To color** | **Use**                    |
|--------------|----------------------------|
| assay        | assay_ontology_term_id     |
| cell_type    | cell_type_ontology_term_id |

File: schema/4.0.0/schema.md [hunk 3]
```diff
@@ -1112,6 +1112,72 @@ See also `default_embedding` in `uns`.
...
 <br>
 
+
+### {column}_colors
+
+<table><tbody>
+  <tr>
+    <th>Key</th>
+      <td>
+        {column}_colors where {column} MUST be the name of a <code>category</code> data type column in <code>obs</code> that<br> is annotated by the data submitter or curator. The following columns that are annotated by CELLxGENE<br> Discover MUST NOT be specified as {category}:<br><br>
```
Comment (jahilton) [inline schema/4.0.0/schema.md:1122]: Discover MUST NOT be specified as {~category~column}:

File: schema/4.0.0/schema.md [hunk 4]
```diff
@@ -1112,6 +1112,72 @@ See also `default_embedding` in `uns`.
...
+  </tr>
+  <tr>
+    <th>Annotator</th>
+    <td>Curator</td>
+  </tr>
+  <tr>
+    <th>Value</th>
+      <td>
+        <code>numpy.ndarray</code>. This MUST be a 1-D array of shape <code>(, c)</code>, where <code>c</code> is greater than or equal to the<br> number of unique categories in the {column} as calculated by:<br><br>
+           <samp>len(anndata.obs.{category}.unique())</samp><br><br>
```
Comment (jahilton) [inline schema/4.0.0/schema.md:1144]: ~category~column

File: schema/4.0.0/schema.md [hunk 5]
```diff
@@ -1112,6 +1112,72 @@ See also `default_embedding` in `uns`.
...
+        The color code at the Nth position in the <code>ndarray</code> corresponds to the Nth category of <samp>anndata.obs.{category}.unique()</samp>.<br><br>For example, if <code>cell_type_ontology_term_id</code> includes two unique categories:<br><br>
```
Comment (jahilton) [inline schema/4.0.0/schema.md:1145]: anndata.obs.{~category~column}.unique()

File: schema/4.0.0/schema.md (hunk 2 above)
Comment (jahilton) [reply schema/4.0.0/schema.md:1131]: I think the added line is sufficient

File: schema/4.0.0/schema.md (hunk 3 above)
Comment (brianraymor) [reply schema/4.0.0/schema.md]: DOH. You'd think that global replace had not been invented.

Addressed in 0f62f4ed2625740ab6cbfc35d6f32123fd2716d4.

File: schema/4.0.0/schema.md (hunk 4 above)
Comment (brianraymor) [reply schema/4.0.0/schema.md]: Addressed in 0f62f4ed2625740ab6cbfc35d6f32123fd2716d4.

File: schema/4.0.0/schema.md (hunk 5 above)
Comment (brianraymor) [reply schema/4.0.0/schema.md]: Addressed in 0f62f4ed2625740ab6cbfc35d6f32123fd2716d4.

**Review by atarashansky — APPROVED** [review]

**Review by jahilton — APPROVED** [review]

**Review by dsadgat — APPROVED** [review]

**Pull request merged**